### PR TITLE
Restrict deletion of default token

### PR DESCRIPTION
**Pull request by pallavi2209**
List all tokens in /v1/user/:id/tokens including default token
No user (admin/non-admin) can delete default token
Cascade delete token on user deletion
Refactor postToken
Refactor deleteUserToken  to restrict default token deletion
Added/updated tests

### CODE REVIEW

**Comment by pallavi2209** [comment]
Sorry with the number of files, most of the changes are added/updated tests.

**Comment by iamigo** [comment]
All of this stuff works with tokens enforced and with tokens not enforced?

**Comment by pallavi2209** [comment]
@iamigo The tests were working fine with token not enforced. Just ran the tests with token enforced, there were some minor updates needed (done and pushed). I have also updated some existing tests to use token and pass with token enforced.
Also tested the use cases in postman. It would be great if you can test some cases at your end too, just to be sure.

File: api/v1/controllers/tokens.js [hunk 1]
```diff
@@ -87,27 +86,22 @@ module.exports = {
...
-      return u.handleError(next, tokenErr, resourceName);
-    }
-
-    // create token to be returned in response.
-    const tokenName = req.swagger.params.queryBody.value.name;
-    const tokenValue = jwtUtil.createToken(tokenName, req.user.name);
+    // get user details from req
+    authUtils.getUser(req)
+    .then((user) => {
+      // create token to be returned in response.
```
Comment (shriramshankar) [inline api/v1/controllers/tokens.js]: created token?

File: api/v1/controllers/userTokens.js [hunk 2]
```diff
@@ -65,63 +65,59 @@ module.exports = {
...
-          .then((o) => {
-            resultObj.dbTime = new Date() - resultObj.reqStartTime;
-            u.logAPI(req, resultObj, o.dataValues);
-            res.status(httpStatus.OK)
-              .json(u.responsify(o, helper, req.method));
-          })
-          .catch((err) => u.handleError(next, err, helper.modelName));
-        });
+      // Default token cannot be deleted
+      if (token.name === token.User.name) {
```
Comment (shriramshankar) [inline api/v1/controllers/userTokens.js]: Do we create a default token with the same name as the username every time a user is created?

File: api/v1/controllers/tokens.js (hunk 1 above)
Comment (pallavi2209) [inline api/v1/controllers/tokens.js]: Sorry, I am confused. Do you mean the "create" in comment? The token is not yet created, so present tense.

File: api/v1/controllers/userTokens.js (hunk 2 above)
Comment (pallavi2209) [inline api/v1/controllers/userTokens.js]: Yes, the logic is in passportconfig in signupStrategy

**Pull request merged**